tests/assets/dynamic-addon-cards.test.js

```diff
@@ -0,0 +1,69 @@
+/* global document */
+const { buildStaticAddonCard } = require('addons-frontend-blog-utils');
+
+const {
+  convertToUnavailableCard,
+  updateAddonCard,
+} = require('../../src/assets/js/dynamic-addon-cards');
+const tabbyAddon = require('../fixtures/amo-api-response-tabby');
+
+describe(__filename, () => {
+  const mockFetch = ({ ok = true, jsonData = {} }) => {
+    return jest.spyOn(global, 'fetch').mockResolvedValue({
+      ok,
+      json: jest.fn().mockResolvedValue(jsonData),
+    });
+  };
+
+  // The code we want to test in this file requires some HTML with static
+  // add-on cards. We can reuse the code that we are using at build time here,
+  // except that we mock the API call to generate a static add-on card in a
+  // predictable manner.
+  const loadStaticAddonCardInDocument = async ({ addon = tabbyAddon } = {}) => {
+    const fetch = mockFetch({ jsonData: addon });
+    const staticCard = await buildStaticAddonCard({ addonId: addon.id });
+    fetch.mockRestore();
+
+    document.body.innerHTML = `<div>${staticCard}</div>`;
+
+    return document.querySelector('.StaticAddonCard');
+  };
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('convertToUnavailableCard', () => {
+    it('changes the state of a card to "unavailable"', async () => {
+      const card = await loadStaticAddonCardInDocument();
+      expect(card.classList).not.toContain('StaticAddonCard--is-unavailable');
+
+      convertToUnavailableCard(card);
+
+      expect(card.classList).toContain('StaticAddonCard--is-unavailable');
+    });
+  });
+
+  describe('updateAddonCard', () => {
+    it('calls the AMO API', async () => {
+      const addon = { ...tabbyAddon };
+      const card = await loadStaticAddonCardInDocument({ addon });
+      const fetch = mockFetch({ jsonData: addon });
+
+      await updateAddonCard(card);
+
+      expect(fetch).toHaveBeenCalledWith(
+        `https://addons.mozilla.org/api/v5/addons/addon/${addon.id}/?lang=en-US&app=firefox`
+      );
+    });
+
+    it('changes the state of a card to "unavailable" when the response is not OK', async () => {
```

Comment:
Probably doesn't matter too much to start with, but should there be a distinction between a different statuses? E.g. a 5xx could be temporary, so should there be a retry?

Reply:
I don't think we want to handle that, or do we?

If user gets 500s from the API, then cards will be "unavailable" but if it's a server hiccup, then refreshing the page should re-update the cards correctly. WDYT?

Reply:
Yeah, it's probably fine to ignore for now. It shouldn't come up too much and if it becomes an issue that would be the point to make things more complicated. 👍